File: api/src/config/env.ts
import dotenv from 'dotenv';

dotenv.config();

const validateEnvVar = (varName: string): string => {
  const value = process.env[varName];
  if (!value) {
    throw new Error(`❌ Environment variable '${varName}' is required but not defined.`);
  }
  return value;
};

const envConfig = {
  DB_PASSWORD: validateEnvVar('DB_PASSWORD'),
  JWT_KEY: validateEnvVar('JWT_KEY'),
  SENDGRID_KEY: validateEnvVar('SENDGRID_KEY'),
} as const;

export default envConfig;

File: api-v2/utils/Env.fs
module Env

let require (key: string) =
    match System.Environment.GetEnvironmentVariable key with
    | null
    | "" -> failwithf "Required environment variable '%s' is not set" key
    | value -> value

let tryAndDefault (key: string) (defaultValue: string) =
    match System.Environment.GetEnvironmentVariable key with
    | null
    | "" -> defaultValue
    | value -> value
